rs/replicated_state/src/canister_state/queues.rs

```diff
@@ -1029,7 +1109,20 @@ impl CanisterQueues {
     ) {
         use Context::*;
 
+        // If this is an inbound response, remember its `originator_reply_callback`, so
+        // we can later produce an `UnknownResponse` for it, when popped.
         let context = reference.context();
+        if let (Inbound, RequestOrResponse::Response(response)) = (context, msg) {
+            assert_eq!(
+                None,
+                self.shed_responses
+                    .insert(reference, response.originator_reply_callback)
```

Comment:
Did we end up making it an invariant or a soft invariant that we have no duplicate response callbacks in queues?

Reply:
"No duplicate callback IDs in inbound queues" is a hard invariant, if that's what you are asking.